d1/david.py

```diff
@@ -0,0 +1,30 @@
+"""
+d1 solution
+"""
+
+"""
+Psuedocode thinking:
+	- have a list [1,2,3,4,5]
+	- python can print list elements doing stack[::1] in order and stack[::-1] in reverse
+	mylist = [1,2,3,4,5]- if we take list length, and read 
+	from front and end mylistand walk towards middle, should be able to create a new list.
+	- struggling to understand the requirement of "must be in place" and how that will work with this code. 
```

Comment:
My understanding of "must be in place" is that you cannot use allocate another memory for a new data structure (variable/stack...etc) besides `stack` and `queue`. It would make a good follow up question though.